client/utils.ts

```diff
@@ -0,0 +1,27 @@
+// Adapts route paths to also include wildcard
+// subroutes under the root level section.
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function pathToRegExp( path: any ) {
+	// Prevents root level double dash urls from being validated.
+	if ( path === '/' ) {
+		return path;
+	}
+	return new RegExp( '^' + path + '(/.*)?$' );
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function debounce( callback: ( ...args: any[] ) => any, timeout: number ) {
+	let timeoutId: number | undefined = undefined;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	return ( ...args: any[] ) => {
+		window.clearTimeout( timeoutId );
+		timeoutId = window.setTimeout( () => {
+			callback( ...args );
+		}, timeout );
+	};
+}
```

Comment:
Added this function and change filed to typescript. The other two functions already existed.